File: tests/expected/bitops.nim

proc main_func() =
  var ands: seq[bool] = @[]
  var ors: seq[bool] = @[]
  var xors: seq[bool] = @[]
  for a in @[false, true]:
    for b in @[false, true]:
      ands.add((a and b))
      ors.add((a or b))
      xors.add((a xor b))
  assert(ands == @[false, false, false, true])
  assert(ors == @[false, true, true, true])
  assert(xors == @[false, true, true, false])
  echo "OK"

proc main() =
  main_func()

main()

File: tests/expected/bitops.kt


fun main_func() {
    var ands: Array<Boolean> = arrayOf()
    var ors: Array<Boolean> = arrayOf()
    var xors: Array<Boolean> = arrayOf()
    for (a in arrayOf(false, true)) {
        for (b in arrayOf(false, true)) {
            ands += a.and(b)
            ors += a.or(b)
            xors += a.xor(b)
        }
    }
    assert(ands == arrayOf(false, false, false, true))
    assert(ors == arrayOf(false, true, true, true))
    assert(xors == arrayOf(false, true, true, false))
    println("OK")
}

fun main(argv: Array<String>) {
    main_func()
}
